### PR TITLE
fix the test failure that to use the clientinfo from oAuthRefreshInterceptor

### PR DESCRIPTION
**Pull request by helenren**

### CODE REVIEW

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/RestClient.java [hunk 1]
```diff
@@ -233,7 +233,7 @@ public String getRefreshToken() {
 	 * @return The client info.
 	 */
 	public ClientInfo getClientInfo() {
-		return clientInfo;
+		return oAuthRefreshInterceptor.clientInfo;
```
Comment (helenren) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/RestClient.java]: the clientInfo is updated for the 2nd request using the access token from authTokenProvider once gotten a 401 return at the first time

Comment (bhariharan) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/RestClient.java]: Maybe we should remove `clientInfo` instance variable from this class and use `oAuthRefreshInterceptor.clientInfo` everywhere? I don't see any reason to keep the instance variable around. It makes it harder to keep the two copies in sync.

Comment (helenren) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/RestClient.java]: agree. Will update the code.

**Comment by wmathurin** [comment]
👍

**Comment by bhariharan** [comment]
👍

**Pull request merged**